sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -101,8 +101,17 @@ import org.apache.spark.sql.types.IntegerType
  */
 object RewriteDistinctAggregates extends Rule[LogicalPlan] {
 
+  private def mayNeedtoRewrite(exprs: Seq[Expression]): Boolean = {
+    val distinctAggs = exprs.flatMap { _.collect {
+      case ae: AggregateExpression if ae.isDistinct => ae
+    }}
+    // We need at least two distinct aggregates for this rule because aggregation
```

Comment:
Let's add a comment that this can produce false positives, i.e. `sum(distinct a)` & `count(distinct a)`

Reply:
suggest to add unit tests for each case of 0, 1, 2+

Reply:
It's hard to check if a rule is applied and does nothing, or a rule is not applied. I'm OK with this simple change.

Reply:
@hvanhovell sure, I'll update the comment.